### ISSUE TITLE
The GeoIP service is not valid.

### ISSUE THREAD

**Issue by hitechqb** [Torann/laravel-geoip#100]
Hi all,
I am using laravel 5.5.
I have config on app.config for Provider and Aliases.
When i call: $location = GeoIP::getLocation(); 
--> Bug: THe GeoIP service is not valid. _(In GeoIP.php line 205)_

Help me!

**Comment by gopalkumar315** [comment]
use the following command link:
php artisan config:clear
php artisan config:cache

**Comment by pdxbug** [comment]
Receiving the same error as here and issues/60. I followed the upgrade instructions:
http://lyften.com/projects/laravel-geoip/doc/upgrade.html

I also have run the following commands:
php artisan config:clear
php artisan config:cache

And continue to get the error message:
php artisan geoip:update

```
In GeoIP.php line 205:
  The GeoIP service is not valid. 
```

When running a curl command to the method:
```
public function testGeoIP()
    {
        $location = GeoIP::getLocation('<a valid IP address>');
        return response()->json($location);
    }
```

I receive:
```
{
    "message": "The GeoIP service is not valid.",
    "exception": "Exception",
    "file": "/my/application/vendor/torann/geoip/src/GeoIP.php",
    "line": 205,
    "trace": [
        {
            "file": "/my/application/vendor/torann/geoip/src/GeoIP.php",
            "line": 169,
            "function": "getService",
            "class": "Torann\\GeoIP\\GeoIP",
            "type": "->"
        },
        {
            "file": "/my/application/vendor/torann/geoip/src/GeoIP.php",
            "line": 114,
            "function": "find",
            "class": "Torann\\GeoIP\\GeoIP",
            "type": "->"
        },
        {
            "file": "/my/application/vendor/laravel/framework/src/Illuminate/Support/Facades/Facade.php",
            "line": 221,
            "function": "getLocation",
            "class": "Torann\\GeoIP\\GeoIP",
            "type": "->"
        },
        {
            "file": "/my/application/app/Http/Controllers/AdPrioritizationController.php",
            "line": 35,
            "function": "__callStatic",
            "class": "Illuminate\\Support\\Facades\\Facade",
            "type": "::"
        },
        {
            "function": "testGeoIP",
            "class": "App\\Http\\Controllers\\AdPrioritizationController",
            "type": "->"
        },
        {
            "file": "/my/application/vendor/laravel/framework/src/Illuminate/Routing/Controller.php",
            "line": 54,
            "function": "call_user_func_array"
        },
        {
            "file": "/my/application/vendor/laravel/framework/src/Illuminate/Routing/ControllerDispatcher.php",
            "line": 45,
            "function": "callAction",
            "class": "Illuminate\\Routing\\Controller",
            "type": "->"
        },
        {
            "file": "/my/application/vendor/laravel/framework/src/Illuminate/Routing/Route.php",
            "line": 212,
            "function": "dispatch",
            "class": "Illuminate\\Routing\\ControllerDispatcher",
            "type": "->"
        },
        {
            "file": "/my/application/vendor/laravel/framework/src/Illuminate/Routing/Route.php",
            "line": 169,
            "function": "runController",
            "class": "Illuminate\\Routing\\Route",
            "type": "->"
        },
        {
            "file": "/my/application/vendor/laravel/framework/src/Illuminate/Routing/Router.php",
            "line": 659,
            "function": "run",
            "class": "Illuminate\\Routing\\Route",
            "type": "->"
        },
        {
            "file": "/my/application/vendor/laravel/framework/src/Illuminate/Routing/Pipeline.php",
            "line": 30,
            "function": "Illuminate\\Routing\\{closure}",
            "class": "Illuminate\\Routing\\Router",
            "type": "->"
        },
        {
            "file": "/my/application/vendor/laravel/framework/src/Illuminate/Routing/Middleware/SubstituteBindings.php",
            "line": 41,
            "function": "Illuminate\\Routing\\{closure}",
            "class": "Illuminate\\Routing\\Pipeline",
            "type": "->"
        },
        {
            "file": "/my/application/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php",
            "line": 151,
            "function": "handle",
            "class": "Illuminate\\Routing\\Middleware\\SubstituteBindings",
            "type": "->"
        },
        {
            "file": "/my/application/vendor/laravel/framework/src/Illuminate/Routing/Pipeline.php",
            "line": 53,
            "function": "Illuminate\\Pipeline\\{closure}",
            "class": "Illuminate\\Pipeline\\Pipeline",
            "type": "->"
        },
        {
            "file": "/my/application/vendor/laravel/framework/src/Illuminate/Routing/Middleware/ThrottleRequests.php",
            "line": 57,
            "function": "Illuminate\\Routing\\{closure}",
            "class": "Illuminate\\Routing\\Pipeline",
            "type": "->"
        },
        {
            "file": "/my/application/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php",
            "line": 151,
            "function": "handle",
            "class": "Illuminate\\Routing\\Middleware\\ThrottleRequests",
            "type": "->"
        },
        {
            "file": "/my/application/vendor/laravel/framework/src/Illuminate/Routing/Pipeline.php",
            "line": 53,
            "function": "Illuminate\\Pipeline\\{closure}",
            "class": "Illuminate\\Pipeline\\Pipeline",
            "type": "->"
        },
        {
            "file": "/my/application/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php",
            "line": 104,
            "function": "Illuminate\\Routing\\{closure}",
            "class": "Illuminate\\Routing\\Pipeline",
            "type": "->"
        },
        {
            "file": "/my/application/vendor/laravel/framework/src/Illuminate/Routing/Router.php",
            "line": 661,
            "function": "then",
            "class": "Illuminate\\Pipeline\\Pipeline",
            "type": "->"
        },
        {
            "file": "/my/application/vendor/laravel/framework/src/Illuminate/Routing/Router.php",
            "line": 636,
            "function": "runRouteWithinStack",
            "class": "Illuminate\\Routing\\Router",
            "type": "->"
        },
        {
            "file": "/my/application/vendor/laravel/framework/src/Illuminate/Routing/Router.php",
            "line": 602,
            "function": "runRoute",
            "class": "Illuminate\\Routing\\Router",
            "type": "->"
        },
        {
            "file": "/my/application/vendor/laravel/framework/src/Illuminate/Routing/Router.php",
            "line": 591,
            "function": "dispatchToRoute",
            "class": "Illuminate\\Routing\\Router",
            "type": "->"
        },
        {
            "file": "/my/application/vendor/laravel/framework/src/Illuminate/Foundation/Http/Kernel.php",
            "line": 176,
            "function": "dispatch",
            "class": "Illuminate\\Routing\\Router",
            "type": "->"
        },
        {
            "file": "/my/application/vendor/laravel/framework/src/Illuminate/Routing/Pipeline.php",
            "line": 30,
            "function": "Illuminate\\Foundation\\Http\\{closure}",
            "class": "Illuminate\\Foundation\\Http\\Kernel",
            "type": "->"
        },
        {
            "file": "/my/application/vendor/barryvdh/laravel-debugbar/src/Middleware/InjectDebugbar.php",
            "line": 65,
            "function": "Illuminate\\Routing\\{closure}",
            "class": "Illuminate\\Routing\\Pipeline",
            "type": "->"
        },
        {
            "file": "/my/application/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php",
            "line": 151,
            "function": "handle",
            "class": "Barryvdh\\Debugbar\\Middleware\\InjectDebugbar",
            "type": "->"
        },
        {
            "file": "/my/application/vendor/laravel/framework/src/Illuminate/Routing/Pipeline.php",
            "line": 53,
            "function": "Illuminate\\Pipeline\\{closure}",
            "class": "Illuminate\\Pipeline\\Pipeline",
            "type": "->"
        },
        {
            "file": "/my/application/vendor/barryvdh/laravel-cors/src/HandleCors.php",
            "line": 36,
            "function": "Illuminate\\Routing\\{closure}",
            "class": "Illuminate\\Routing\\Pipeline",
            "type": "->"
        },
        {
            "file": "/my/application/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php",
            "line": 151,
            "function": "handle",
            "class": "Barryvdh\\Cors\\HandleCors",
            "type": "->"
        },
        {
            "file": "/my/application/vendor/laravel/framework/src/Illuminate/Routing/Pipeline.php",
            "line": 53,
            "function": "Illuminate\\Pipeline\\{closure}",
            "class": "Illuminate\\Pipeline\\Pipeline",
            "type": "->"
        },
        {
            "file": "/my/application/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/CheckForMaintenanceMode.php",
            "line": 51,
            "function": "Illuminate\\Routing\\{closure}",
            "class": "Illuminate\\Routing\\Pipeline",
            "type": "->"
        },
        {
            "file": "/my/application/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php",
            "line": 151,
            "function": "handle",
            "class": "Illuminate\\Foundation\\Http\\Middleware\\CheckForMaintenanceMode",
            "type": "->"
        },
        {
            "file": "/my/application/vendor/laravel/framework/src/Illuminate/Routing/Pipeline.php",
            "line": 53,
            "function": "Illuminate\\Pipeline\\{closure}",
            "class": "Illuminate\\Pipeline\\Pipeline",
            "type": "->"
        },
        {
            "file": "/my/application/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php",
            "line": 104,
            "function": "Illuminate\\Routing\\{closure}",
            "class": "Illuminate\\Routing\\Pipeline",
            "type": "->"
        },
        {
            "file": "/my/application/vendor/laravel/framework/src/Illuminate/Foundation/Http/Kernel.php",
            "line": 151,
            "function": "then",
            "class": "Illuminate\\Pipeline\\Pipeline",
            "type": "->"
        },
        {
            "file": "/my/application/vendor/laravel/framework/src/Illuminate/Foundation/Http/Kernel.php",
            "line": 116,
            "function": "sendRequestThroughRouter",
            "class": "Illuminate\\Foundation\\Http\\Kernel",
            "type": "->"
        },
        {
            "file": "/my/application/public/index.php",
            "line": 53,
            "function": "handle",
            "class": "Illuminate\\Foundation\\Http\\Kernel",
            "type": "->"
        }
    ]
}
```

Let me know what else you might need to help diagnose. Thanks.

**Comment by 13Aks13** [comment]
I have the same error for Laravel 5.5  I have config on app.config for Provider and Aliases 
 prov ->   Torann\GeoIP\GeoIPServiceProvider::class,
 alias-> 'GeoIP' => Torann\GeoIP\Facades\GeoIP::class,

Exception: The GeoIP service is not valid. (In GeoIP.php line 205)

Anybody have resolved for this error?

**Comment by bhavinjr** [comment]
I had the same problem and I fixed it up.
just update config/geoip.php file 
you can check here `https://github.com/Torann/laravel-geoip/blob/master/config/geoip.php`

**Comment by liyu001989** [comment]
I solve this after publish config
```
php artisan vendor:publish --provider="Torann\GeoIP\GeoIPServiceProvider" --tag=config
```

**Comment by bhavinjr** [comment]
@liyu001989 if someone already publishes config file with the older version then it won't work
and `geoip.php` file is updated have to update `geoip.php` file

**Comment by summerblue** [comment]
@liyu001989 Same problem, awesome fix.

**Comment by acidjazz** [comment]
I don't really agree on this being a "fix", shouldn't laravel packages not require their config to be published? this stop from allowing packages that use packages/etc.

**Comment by acidjazz** [comment]
submitted a PR that fixes this #119
